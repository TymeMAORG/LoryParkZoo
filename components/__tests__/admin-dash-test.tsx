import React from 'react';
import renderer from 'react-test-renderer';
import AdminDashboard from '../../app/(tabs)/admin/index';

describe('AdminDashboard', () => {
    it('renders the correct title and date', () => {
        const tree = renderer.create(<AdminDashboard />).toJSON();
        expect(tree).toMatchSnapshot();

        const instance = renderer.create(<AdminDashboard />).root;
        const title = instance.findByProps({ children: 'Admin Dashboard' });
        expect(title).toBeTruthy();

        const date = new Date();
        const formattedDate = date.toLocaleDateString('en-US', {
            weekday: 'long',
            month: 'long',
            day: 'numeric',
        });

        const dateText = instance.findByProps({ children: formattedDate });
        expect(dateText).toBeTruthy();
    });
});
