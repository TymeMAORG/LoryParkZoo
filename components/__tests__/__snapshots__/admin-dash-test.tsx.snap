// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminDashboard renders the correct title and date 1`] = `
<View
  style={
    {
      "flex": 1,
      "padding": 16,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    Admin Dashboard
  </Text>
  <Text
    style={
      {
        "fontSize": 18,
        "marginTop": 8,
      }
    }
  >
    Monday, October 21
  </Text>
</View>
`;

exports[`AdminDashboard renders the welcome message with the correct username 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#f5f5f5",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View>
    <Text
      style={
        {
          "fontSize": 24,
          "fontWeight": "bold",
          "marginBottom": 20,
        }
      }
    >
      Welcome Admin, 
      admin
      !
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "elevation": 3,
            "flex": 1,
            "marginHorizontal": 5,
            "padding": 15,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
      >
        <Text
          style={
            {
              "color": "#2196F3",
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          15
        </Text>
        <Text
          style={
            {
              "color": "#666",
              "fontSize": 12,
              "marginTop": 5,
            }
          }
        >
          Total Staff
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "elevation": 3,
            "flex": 1,
            "marginHorizontal": 5,
            "padding": 15,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
      >
        <Text
          style={
            {
              "color": "#2196F3",
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          45
        </Text>
        <Text
          style={
            {
              "color": "#666",
              "fontSize": 12,
              "marginTop": 5,
            }
          }
        >
          Animals
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "elevation": 3,
            "flex": 1,
            "marginHorizontal": 5,
            "padding": 15,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
      >
        <Text
          style={
            {
              "color": "#2196F3",
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          8
        </Text>
        <Text
          style={
            {
              "color": "#666",
              "fontSize": 12,
              "marginTop": 5,
            }
          }
        >
          Sections
        </Text>
      </View>
    </View>
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 15,
          "marginTop": 10,
        }
      }
    >
      Recent Activity
    </Text>
    <View
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "padding": 15,
        }
      }
    >
      <View
        style={
          {
            "borderBottomColor": "#eee",
            "borderBottomWidth": 1,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 16,
            }
          }
        >
          Sarah submitted daily checklist
        </Text>
        <Text
          style={
            {
              "color": "#666",
              "fontSize": 12,
              "marginTop": 5,
            }
          }
        >
          2 hours ago
        </Text>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#eee",
            "borderBottomWidth": 1,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 16,
            }
          }
        >
          John updated feeding schedule
        </Text>
        <Text
          style={
            {
              "color": "#666",
              "fontSize": 12,
              "marginTop": 5,
            }
          }
        >
          4 hours ago
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`AdminDashboard renders welcome message with username 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#f5f5f5",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View>
    <Text
      style={
        {
          "fontSize": 24,
          "fontWeight": "bold",
          "marginBottom": 20,
        }
      }
    >
      Welcome Admin, 
      AdminUser
      !
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "elevation": 3,
            "flex": 1,
            "marginHorizontal": 5,
            "padding": 15,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
      >
        <Text
          style={
            {
              "color": "#2196F3",
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          15
        </Text>
        <Text
          style={
            {
              "color": "#666",
              "fontSize": 12,
              "marginTop": 5,
            }
          }
        >
          Total Staff
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "elevation": 3,
            "flex": 1,
            "marginHorizontal": 5,
            "padding": 15,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
      >
        <Text
          style={
            {
              "color": "#2196F3",
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          45
        </Text>
        <Text
          style={
            {
              "color": "#666",
              "fontSize": 12,
              "marginTop": 5,
            }
          }
        >
          Animals
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "elevation": 3,
            "flex": 1,
            "marginHorizontal": 5,
            "padding": 15,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
      >
        <Text
          style={
            {
              "color": "#2196F3",
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          8
        </Text>
        <Text
          style={
            {
              "color": "#666",
              "fontSize": 12,
              "marginTop": 5,
            }
          }
        >
          Sections
        </Text>
      </View>
    </View>
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 15,
          "marginTop": 10,
        }
      }
    >
      Recent Activity
    </Text>
    <View
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "padding": 15,
        }
      }
    >
      <View
        style={
          {
            "borderBottomColor": "#eee",
            "borderBottomWidth": 1,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 16,
            }
          }
        >
          Sarah submitted daily checklist
        </Text>
        <Text
          style={
            {
              "color": "#666",
              "fontSize": 12,
              "marginTop": 5,
            }
          }
        >
          2 hours ago
        </Text>
      </View>
      <View
        style={
          {
            "borderBottomColor": "#eee",
            "borderBottomWidth": 1,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 16,
            }
          }
        >
          John updated feeding schedule
        </Text>
        <Text
          style={
            {
              "color": "#666",
              "fontSize": 12,
              "marginTop": 5,
            }
          }
        >
          4 hours ago
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
